### PR TITLE
[5G: MVC][component: agw][type: enhancement] AMF Base Timer Changes

### PR DESCRIPTION
**Pull request by hkshiv1**
Signed-off-by: hkshiv1 <shivakumar.hk@acldigital.com>

## Summary

AMF Timers T3560, T3570, T3550 code changes

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: lte/gateway/c/oai/tasks/amf/amf_Security_Mode.cpp [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/*****************************************************************************
+
+  Source      amf_Security_Mode.cpp
+
+  Version     0.1
```
Comment (sureshgorijavolu) [inline lte/gateway/c/oai/tasks/amf/amf_Security_Mode.cpp:17]: take out version, subsystem, author & product

File: lte/gateway/c/oai/tasks/amf/amf_Security_Mode.cpp [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+  }
+
+  nas_amf_smc_proc_t* smc_proc = get_nas5g_common_procedure_smc(amf_ctx);
+
+  if (smc_proc) {
+    /*
+     * Stop timer T3560
+     */
+
+    OAILOG_ERROR(
```
Comment (sureshgorijavolu) [inline lte/gateway/c/oai/tasks/amf/amf_Security_Mode.cpp:89]: this should be an info level log statement.

File: lte/gateway/c/oai/tasks/amf/amf_Security_Mode.cpp [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        LOG_AMF_APP, "Timer: Calling SMC MODE stop timer with id = %d\n",
+        smc_proc->T3560.id);
+    stop_timer(&amf_app_task_zmq_ctx, smc_proc->T3560.id);
+    OAILOG_ERROR(LOG_AMF_APP, "Timer: After stopping SMC MODE timer \n");
```
Comment (sureshgorijavolu) [inline lte/gateway/c/oai/tasks/amf/amf_Security_Mode.cpp:93]: is this a DEBUG level log statement?

File: lte/gateway/c/oai/tasks/amf/amf_Security_Mode.cpp [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+      amf_sap.u.amf_reg.ctx           = amf_ctx;
+      amf_sap.u.amf_reg.notify        = true;
+      amf_sap.u.amf_reg.free_proc     = true;
+      amf_sap.u.amf_reg.u.common_proc = &smc_proc->amf_com_proc;
+
+      amf_ctx->_security.kenb_ul_count = amf_ctx->_security.ul_count;
+      amf_ctx_set_attribute_valid(amf_ctx, AMF_CTXT_MEMBER_SECURITY);
+      rc = amf_sap_seq.amf_sap_send(&amf_sap);
+    }
+    /* Nothing to do in
```
Comment (sureshgorijavolu) [inline lte/gateway/c/oai/tasks/amf/amf_Security_Mode.cpp:112]: can you modify the comment? it is not conveying clear message.

File: lte/gateway/c/oai/tasks/amf/amf_Security_Mode.cpp [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+     * Calling SMC response success and triggering registration accept message*/
+    amf_registration_procedure::amf_registration_success_security_cb(amf_ctx);
+    // amf_registration_success_security_cb(amf_ctx);
```
Comment (sureshgorijavolu) [inline lte/gateway/c/oai/tasks/amf/amf_Security_Mode.cpp:115]: if you no longer need this dead code, take it out.

File: lte/gateway/c/oai/tasks/amf/amf_identity.cpp [hunk 6]
```diff
@@ -0,0 +1,373 @@
...
+    imei.u.num.snr1   = msg->mobile_identity.imei.identity_digit9;
+    imei.u.num.snr2   = msg->mobile_identity.imei.identity_digit10;
+    imei.u.num.snr3   = msg->mobile_identity.imei.identity_digit11;
+    imei.u.num.snr4   = msg->mobile_identity.imei.identity_digit12;
+    imei.u.num.snr5   = msg->mobile_identity.imei.identity_digit13;
+    imei.u.num.snr6   = msg->mobile_identity.imei.identity_digit14;
+    imei.u.num.cdsd   = msg->mobile_identity.imei.identity_digit15;
+    imei.u.num.parity = msg->mobile_identity.imei.odd_even;
+  }
+#if 0
```
Comment (sureshgorijavolu) [inline lte/gateway/c/oai/tasks/amf/amf_identity.cpp:143]: we do not need code that does not compile.

**Comment by ulaskozat** [comment]
Can you please indicate on which other PR(s) this PR is dependent?

**Comment by ulaskozat** [comment]
remove all the commented code out if they are not used. clearly list to do and have place holder functions instead.

File: lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h [hunk 7]
```diff
@@ -0,0 +1,1348 @@
...
+
+typedef struct qos_flow_request_list_s {
+  ie_header hdr; 
+  qos_flow_setup_request_item qos_flow_req_item;
+
+} qos_flow_request_list;
+#endif  // pdu_res_set_change
+// Resource relese request and response
+// typedef enum radio_network_layer_cause_e
+#if 0
```
Comment (ssanadhya) [inline lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h:184]: This is effectively dead code, please remove!

File: lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h [hunk 8]
```diff
@@ -0,0 +1,1348 @@
...
+//=============================
+//#if 1
+
+typedef struct ie_header_t {
+  uint16_t protocol_id;
+  uint8_t criticality;
+  uint8_t pad;
+} ie_header;
+
+#if 0
```
Comment (ssanadhya) [inline lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h:124]: This is effectively dead code, please remove!

File: lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h [hunk 9]
```diff
@@ -0,0 +1,1348 @@
...
+#define NAS5G_MESSAGE_CONTAINER_MAXIMUM_LENGTH 253
+
+/* Timer structure */
+typedef struct amf_app_timer_s {
+  long id;  /* The timer identifier                 */
+  long sec; /* The timer interval value in seconds  */
+} amf_app_timer_t;
+
+//=============================
+//#if 1
```
Comment (ssanadhya) [inline lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h:116]: Remove

File: lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h [hunk 10]
```diff
@@ -0,0 +1,1348 @@
...
+typedef struct pdu_session_resource_released_item_s {
+  PDUSessionIdentityMsg pdu_session_id;
+  pdu_session_resource_release_response_transfer release_response_transfer;
+} pdu_session_resource_released_item;
+
+typedef struct pdu_session_resource_released_list_s {
+  pdu_session_resource_released_item released_item;
+} pdu_session_resource_released_list;
+
+#if 0
```
Comment (ssanadhya) [inline lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h:407]: Remove dead code

**Review by ssanadhya — CHANGES_REQUESTED** [review]
Please remove all code inside `#if 0` blocks.

**Pull request closed without merging**